### PR TITLE
64-bit fixes

### PR DESCRIPTION
**Pull request by GravisZro**
### Description
* Switch all the (u)long types to (u)int32_t where appropriate.
* ~~Cleaned up the Win32 API arguments~~ (#354 beat me to it by like 12 minutes)
* ~~Fixes the libmve code (failed to dereference the array)~~ (moved into #359)
* ~~Fixes the network code which was using the incorrect size due to use of "long"~~ (moved into #356)

## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes (return types are now consistent with the data format)
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

**Comment by Lgt2x** [comment]
Any overlap with #354 ?

**Comment by GravisZro** [comment]
> Any overlap with #354 ?

Yes. I suppose I'll remove the overlap.

**Comment by pzychotic** [comment]
What a timing with my stuff :)

> Any overlap with #354 ?

Most stuff I touched in https://github.com/DescentDevelopers/Descent3/pull/354/commits/6ecafd9e68a47c9bc367cdaf8fde7aae98acee75 is done in a slightly different way.

**Comment by GravisZro** [comment]
> What a timing with my stuff :)

I swear it's like any time I make a contribution that someone just MUST get in and do the same thing. *sigh*

File: third_party/stb/stb_image_write.h [hunk 1]
```diff
@@ -292,8 +292,8 @@ static void stbi__stdio_write(void *context, void *data, int size)
 #else
 #define STBIW_EXTERN extern
 #endif
-STBIW_EXTERN __declspec(dllimport) int __stdcall MultiByteToWideChar(unsigned int cp, unsigned long flags, const char *str, int cbmb, wchar_t *widestr, int cchwide);
-STBIW_EXTERN __declspec(dllimport) int __stdcall WideCharToMultiByte(unsigned int cp, unsigned long flags, const wchar_t *widestr, int cchwide, char *str, int cbmb, const char *defchar, int *used_default);
+STBIW_EXTERN __declspec(dllimport) int __stdcall MultiByteToWideChar(unsigned int cp, uint32_t flags, const char *str, int cbmb, wchar_t *widestr, int cchwide);
+STBIW_EXTERN __declspec(dllimport) int __stdcall WideCharToMultiByte(unsigned int cp, uint32_t flags, const wchar_t *widestr, int cchwide, char *str, int cbmb, const char *defchar, int *used_default);
```
Comment (winterheart) [inline third_party/stb/stb_image_write.h:296]: This is external library, no need change it.

**Comment by GravisZro** [comment]
@winterheart issues resolved.

**Review by Lgt2x — APPROVED** [review]
Great work, thanks! Diff looks good, I'll give it a go on Windows and let @winterheart approve additionally. Good idea to split it in multiple PR for easier reviewing

**Review by winterheart — APPROVED** [review]

**Pull request merged**